teos/teosd.py

```diff
@@ -68,171 +56,273 @@ def get_config(command_line_conf, data_dir):
+            if len(missed_blocks_watcher) == 0 and len(missed_blocks_responder) != 0:
+                Builder.populate_block_queue(self.watcher.responder.block_queue, missed_blocks_responder)
+                self.watcher.responder.block_queue.join()
+
+            elif len(missed_blocks_responder) == 0 and len(missed_blocks_watcher) != 0:
+                Builder.populate_block_queue(self.watcher.block_queue, missed_blocks_watcher)
+                self.watcher.block_queue.join()
+
+            # Otherwise they need to be updated at the same time, block by block
+            elif len(missed_blocks_responder) != 0 and len(missed_blocks_watcher) != 0:
+                Builder.update_states(self.watcher, missed_blocks_watcher, missed_blocks_responder)
+
+        # Fire ChainMonitor
+        # FIXME: 92-block-data-during-bootstrap-db
+        self.chain_monitor.monitor_chain()
+
+    def start_services(self):
+        """Readies the tower by setting up signal handling, and starting all the services."""
+        signal(SIGINT, self.handle_signals)
+        signal(SIGTERM, self.handle_signals)
+        signal(SIGQUIT, self.handle_signals)
+
+        # Start the internal API
+        self.internal_api.rpc_server.start()
+        logger.info(f"Internal API initialized. Serving at {INTERNAL_API_ENDPOINT}")
+
+        # Start the rpc process
+        self.rpc_process.start()
+
+        # Start the public API server
+        api_endpoint = f"{self.config.get('API_BIND')}:{self.config.get('API_PORT')}"
+        if self.config.get("WSGI") == "gunicorn":
+            # FIXME: We may like to add workers depending on a config value
+            self.api_popen = subprocess.Popen(
+                [
+                    "gunicorn",
+                    f"--bind={api_endpoint}",
+                    f"teos.api:serve(internal_api_endpoint='{INTERNAL_API_ENDPOINT}', "
+                    f"endpoint='{api_endpoint}', min_to_self_delay='{self.config.get('MIN_TO_SELF_DELAY')}', "
+                    f"log_file='{self.config.get('LOG_FILE')}')",
+                ]
             )
+        else:
+            self.api_process = multiprocessing.Process(
+                target=api.serve,
+                kwargs={
+                    "internal_api_endpoint": INTERNAL_API_ENDPOINT,
+                    "endpoint": api_endpoint,
+                    "min_to_self_delay": self.config.get("MIN_TO_SELF_DELAY"),
+                    "log_file": self.config.get("LOG_FILE"),
+                    "auto_run": True,
+                },
+            )
+            self.api_process.start()
 
-            watcher_appointments_data = db_manager.load_watcher_appointments()
-            responder_trackers_data = db_manager.load_responder_trackers()
+    def handle_signals(self, signum, frame):
+        """Handles signals by initiating a graceful shutdown."""
+        logger.info(f"Signal {signum} received. Stopping")
```

Comment:
I don't see the point of this log. May be useful for debug, but I don't think it is as info.

Reply:
Changed to debug in 04d6ef9.